common/src/main/java/io/netty/util/concurrent/AbstractScheduledEventExecutor.java

```diff
@@ -269,29 +256,17 @@ public void run() {
 
     final void removeScheduled(final ScheduledFutureTask<?> task) {
         if (inEventLoop()) {
-            removedSchedule0(task);
+            scheduledTaskQueue().removeTyped(task);
```

Comment:
Should we better keep the optimisation we had before (peek first).

Reply:
I thought the peek first addition was just to determine whether or not `minimumDelayScheduledTaskRemoved` should be called? From cursory glance at `DefaultPriorityQueue` impl it's not clear to me that it would be faster, but happy to change it (I may be missing something)

Reply:
maybe you are right... never mind